packages/aws-cdk/lib/api/logs/logs-monitor.ts

```diff
@@ -0,0 +1,248 @@
+    this.active = true;
+  }
+
+  /**
+   * deactivates the monitor so no new events are read
+   * use case for this is when we are in the middle of performing a deployment
+   * and don't want to interweave all the logs together with the CFN
+   * deployment logs
+   *
+   * Also resets the start time to be when the new deployment was triggered
+   * and clears the list of tracked log groups
+   */
+  public deactivate(): void {
+    this.active = false;
+    this.startTime = Date.now();
+    this.logGroups.clear();
+  }
+
+  /**
+   * Adds CloudWatch log groups to read log events from.
+   * Since we could be watching multiple stacks that deploy to
+   * multiple environments (account+region), we need to store a list of log groups
+   * per env along with the SDK object that has access to read from
+   * that AWS account.
+   */
+  public addLogGroups(options: AddLogGroupOptions): void {
+    const env = `${options.account}:${options.region}`;
+    const groups = new Map<string, number>();
+    options.groups.forEach(group => {
+      groups.set(group, this.startTime);
+    });
+    const logGroup = this.logGroups.get(env);
+    if (logGroup) {
+      groups.forEach((time, group) => {
+        logGroup.groups.set(group, time);
+      });
+    } else {
+      this.logGroups.set(env, {
+        sdk: options.sdk,
+        groups,
+      });
+    }
+  }
+
+  private scheduleNextTick(): void {
+    setTimeout(() => void(this.tick()), SLEEP);
+  }
+
+  private async tick(): Promise<void> {
+    if (!this.active) {
+      this.scheduleNextTick();
+      return;
+    }
+    try {
+      const events = flatten(await this.readNewEvents());
+      events.forEach(event => {
+        this.print(event);
+      });
+    } catch (e) {
+      error('Error occurred while monitoring logs: %s', e);
+    }
+    this.scheduleNextTick();
+  }
+
+  /**
+   * Print out a cloudwatch event
+   */
+  private print(event: CloudWatchLogEvent): void {
+    print(util.format('[%s] %s %s',
+      colors.blue(event.logGroup),
+      colors.yellow(event.timestamp.toLocaleTimeString()),
+      event.message.trim()));
+  }
+
+  /**
+   * Reads all new log events from a set of CloudWatch Log Groups
+   * in parallel
+   */
+  private async readNewEvents(): Promise<Array<Array<CloudWatchLogEvent>>> {
+    const promises: Array<Promise<Array<CloudWatchLogEvent>>> = [];
+    for (const config of this.logGroups.values()) {
+      const sdk = config.sdk;
+      for (const group of config.groups.keys()) {
+        promises.push(this.readEventsFromLogGroup(config.groups, sdk, group));
+      }
+    }
+    return Promise.all(promises);
+  }
+
+  /**
+   * Reads all new log events from a CloudWatch Log Group
+   * starting when the hotswap was triggered
+   *
+   * Only prints out events that have not been printed already
+   */
+  private async readEventsFromLogGroup(logGroupConfig: Map<string, number>, sdk: ISDK, logGroupName: string): Promise<CloudWatchLogEvent[]> {
```

Comment:
Let's be consistent here:

```suggestion
  private async readEventsFromLogGroup(logGroupConfig: Map<string, number>, sdk: ISDK, logGroupName: string): Promise<Array<CloudWatchLogEvent>> {
```